android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java

```diff
@@ -78,101 +80,145 @@ public void resetServices(TxService mTxService, KeyringService mKeyringService,
             BlockchainRegistry mBlockchainRegistry, JsonRpcService mJsonRpcService,
             EthTxManagerProxy mEthTxManagerProxy, SolanaTxManagerProxy mSolanaTxManagerProxy,
             BraveWalletService mBraveWalletService, AssetRatioService mAssetRatioService) {
-        this.mTxService = mTxService;
-        this.mKeyringService = mKeyringService;
-        this.mBlockchainRegistry = mBlockchainRegistry;
-        this.mJsonRpcService = mJsonRpcService;
-        this.mEthTxManagerProxy = mEthTxManagerProxy;
-        this.mSolanaTxManagerProxy = mSolanaTxManagerProxy;
-        this.mBraveWalletService = mBraveWalletService;
-        this.mAssetRatioService = mAssetRatioService;
-        mPendingTxHelper.setTxService(mTxService);
-        mNetworkModel.resetServices(mJsonRpcService);
+        synchronized (mLock) {
```

Comment:
so if other methods are being executed and has the lock then `resetServices` will have to wait till the lock is released? I guess it can cause the same problem?

I guess since the model is being created on main thread inside `onConnectionError` and `finishNativeInitialization` so it's thread safe so just setting the services to null via reset and null checks should be a quick and safe solution as we don't want the `resetServices` method to wait for the lock to release to nullify the services.

Reply:
it could be that we can get rid of the lock, but I don't see how it can slow down. What is the downside of `resetServices` to wait while the lock from other methods execution be unlocked? Almost all the methods are fast and are not executed frequently plus the connection drop happens rare. I don't think there will be any slowness at all, but the lock gives us more protection theoretically.